tests/pipeline/test_refresh_modes.py

```diff
@@ -0,0 +1,266 @@
+        result = client.execute_sql("SELECT id FROM some_data_1 ORDER BY id")
+    assert result == [(1,), (2,)]
+
+    # Loaded state contains only keys created in second run
+    with pipeline.destination_client() as dest_client:
+        destination_state = load_pipeline_state_from_destination(
+            pipeline.pipeline_name, dest_client  # type: ignore[arg-type]
+        )
+    assert destination_state["sources"]["my_source"]["resources"] == {
+        "some_data_1": {"resource_key_3": "resource_value_3"},
+    }
+
+
+def test_refresh_drop_tables():
+    first_run = True
+
+    @dlt.source
+    def my_source():
+        @dlt.resource
+        def some_data_1():
+            # Set some source and resource state
+            if first_run:
+                dlt.state()["source_key_1"] = "source_value_1"
+                resource_state("some_data_1")["resource_key_1"] = "resource_value_1"
+                resource_state("some_data_1")["resource_key_2"] = "resource_value_2"
+            else:
+                # State is cleared for all resources on second run
+                assert "source_key_1" in dlt.state()
+                assert "source_key_2" in dlt.state()
+                assert "source_key_3" in dlt.state()
+                # Resource 3 is not wiped
+                assert dlt.state()["resources"] == {
+                    "some_data_3": {"resource_key_5": "resource_value_5"}
+                }
+            yield {"id": 1, "name": "John"}
+            yield {"id": 2, "name": "Jane"}
+
+        @dlt.resource
+        def some_data_2():
+            if first_run:
+                dlt.state()["source_key_2"] = "source_value_2"
+                resource_state("some_data_2")["resource_key_3"] = "resource_value_3"
+                resource_state("some_data_2")["resource_key_4"] = "resource_value_4"
+            else:
+                resource_state("some_data_2")["resource_key_6"] = "resource_value_6"
+            yield {"id": 3, "name": "Joe"}
+            yield {"id": 4, "name": "Jill"}
+
+        @dlt.resource
+        def some_data_3():
+            if first_run:
+                dlt.state()["source_key_3"] = "source_value_3"
+                resource_state("some_data_3")["resource_key_5"] = "resource_value_5"
+            yield {"id": 5, "name": "Jack"}
+            yield {"id": 6, "name": "Jill"}
+
+        return [some_data_1, some_data_2, some_data_3]
+
+    # First run pipeline with load to destination so tables are created
+    pipeline = dlt.pipeline(
+        "refresh_full_test",
+        destination="duckdb",
+        refresh="drop_tables",
+        dataset_name="refresh_full_test",
+    )
+
+    info = pipeline.run(my_source())
+    assert_load_info(info)
+
+    # Second run of pipeline with only selected resources
+    first_run = False
+    info = pipeline.run(my_source().with_resources("some_data_1", "some_data_2"))
+
+    # Confirm resource tables not selected on second run got wiped
```

Comment:
w8 - in this mode tables not selected should not be touched. I think that just the comment is outdated